packages/amplify-cli-core/src/feature-flags/featureFlags.ts

```diff
@@ -158,7 +159,11 @@ export class FeatureFlags {
     FeatureFlags.ensureInitialized();
 
     if (!envNames) {
-      throw new Error("'envNames' argument is required");
+      // this is an internal issue, we either couldn't load the environment names
+      // or their configuration is invalid, further troubleshooting is needed
+      throw amplifyFaultWithTroubleshootingLink('ConfigurationFault', {
+        message: 'Environment names could not be loaded or were not provided.',
+      });
```

Comment:
do you know why do we have `amplifyFaultWithTroubleshootingLink` factory method instead of defaulting the `link` in `AmplifyError` ctor ?
The naming of `amplifyFaultWithTroubleshootingLink` is quite unfortunate. If `a` was capital letter it would look like subtype of `AmplifyError`. There's a verb missing there.

Reply:
I have proposed a slight refactor of our error handling classes and utilities (we can use decorator or factory); but it is a second priority to this at the moment. We are concluding work with the error handling migration, so it became a lower priority (we have already completed the work for the top 6 commands, which was the original goal); I will refactor if I have extra time this sprint

Reply:
here is the task for follow up https://app.asana.com/0/1203051621673505/1203051593370893/f

Reply:
Yup. It's a legacy from POV of this PR. However, if we spread coverage the follow up might be big. so good to think about doing that with next error handler migration piece.

Reply:
Btw I noticed on the other PR that amplifyFaultWithTroubleshootingLink is deprecated.